### PR TITLE
Concert venue form

### PR DESCRIPTION
**Pull request by Jessica-Mach**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -74,6 +74,10 @@
 			<artifactId>frontend-maven-plugin</artifactId>
 			<version>1.7.6</version>
 		</dependency>
+		<dependency> 
+    <groupId>org.springframework.boot</groupId> 
+    <artifactId>spring-boot-starter-validation</artifactId> 
+</dependency>
```
Comment (bkincart) [inline pom.xml]: Please fix indentation here!

File: src/main/frontend/components/ConcertVenueForm.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    zipCode:"",
+    capacity:"",
+    description:"",
+    imgUrl:"",
+    phoneNumber:""
+  })
+
+  const [errors, setErrors]= useState({})
+  const [shouldRedirect, setShouldRedirect] = useState(false)
+  const [venueId, setVenueId]= useState(null)
```
Comment (bkincart) [inline src/main/frontend/components/ConcertVenueForm.js]: Let's clean up some whitespace here: add a space after each key in your `formPayload` object lines 11-19 and 70-78, and before the equals signs ahead of `useState`

File: src/main/frontend/components/ConcertVenueForm.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        <label htmlFor="phoneNumber">Phone Number:</label>
+        <input
+          name= "phoneNumber"
+          id="phoneNumber"
+          type="tel"
+          pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
+          value={formPayload.phoneNumber}
+          onChange={handleInputChange}
+        />
+        <small>Format: 123-456-7890</small>
```
Comment (bkincart) [inline src/main/frontend/components/ConcertVenueForm.js:166]: Nice user experience!

File: src/main/frontend/components/ConcertVenueForm.js [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+          onChange={handleInputChange}
+        />
+      </div>
+
+      <input className="button" type="submit" value="Submit" />
+    </form>
+  )
+}
+
+export default ConcertVenueForm
```
Comment (bkincart) [inline src/main/frontend/components/ConcertVenueForm.js:208]: Very nicely done here :)

File: src/main/frontend/components/NavBar.js [hunk 5]
```diff
@@ -12,6 +12,8 @@ const NavBar = (props) => {
           <Redirect to="/concert-venues" />
         </Route>
         <Route exact path="/concert-venues" component={VenueIndex} />
+        <Route exact path="/concert-venues/new" component={ConcertVenueForm} />
+        <Route exact path="/concert-venues/:id" />
```
Comment (bkincart) [inline src/main/frontend/components/NavBar.js]: Let's remove this `show` path from this PR!

File: src/main/frontend/components/StateDropdown.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    <option value="WI">Wisconsin</option>
+    <option value="WY">Wyoming</option>
+  </select>				
+
+   </div>
+ 
+ )
+
+}
+export default StateDropdown
```
Comment (bkincart) [inline src/main/frontend/components/StateDropdown.js:65]: Love that you extracted this to its own component :) 

One refactor: create an object of the data with abbreviation:state key-value pairs, e.g.:
```
{
  "WI": "Wisconsin",
  "WY": "Wyoming"
}
```

And then `map` to create the `option` tags.

At a minimum, though, let's fix up indentation and remove the whitespace on lines 62, 64, and 66 and add an empty line after 67 :)

File: src/main/java/com/launchacademy/reviews/controllers/VenueController.java [hunk 7]
```diff
@@ -43,4 +49,17 @@ public VenueController(VenueService venueService) {
...
+  public ResponseEntity<Object> createVenue(@Valid @RequestBody Venue venue, BindingResult bindingResult ) {
+    if (bindingResult.hasErrors()) {
+      Map<String, String> errorMap = ErrorDetails.populateErrors(bindingResult);
+      ErrorDetails errorDetails = new ErrorDetails(new Date(), errorMap, "Validation Failed");
+      return new ResponseEntity<Object>(errorDetails, HttpStatus.UNPROCESSABLE_ENTITY);
+    }
+    Venue newVenue = venueService.save(venue);
+    Map<String, Venue> venueMap = new HashMap<>();
+    venueMap.put("venue", newVenue);
+    return new ResponseEntity<Object>(venueMap, HttpStatus.CREATED);
```
Comment (bkincart) [inline src/main/java/com/launchacademy/reviews/controllers/VenueController.java:62]: good!

File: src/main/java/com/launchacademy/reviews/errorHandlers/ErrorDetails.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+  public static Map<String, String> populateErrors(BindingResult bindingResult) {
+    Map<String,String> errorMap = new HashMap<>();
+    List<FieldError> fieldErrorList = bindingResult.getFieldErrors();
+    for (FieldError fieldError: fieldErrorList ){
+      errorMap.put(fieldError.getField(), fieldError.getDefaultMessage());
+      System.out.println(fieldError.getField());
+      System.out.println(fieldError.getDefaultMessage());
+    }
+    return errorMap;
```
Comment (bkincart) [inline src/main/java/com/launchacademy/reviews/errorHandlers/ErrorDetails.java:33]: Once again, nice extracting of this logic!

File: src/main/java/com/launchacademy/reviews/models/Venue.java [hunk 9]
```diff
@@ -49,7 +61,9 @@
   @Column(name = "description")
   private String description;
 
+  @URL(regexp = "https?:\\/\\/(www\\.)?[-a-zA-Z0-9@:%._\\+~#=]{1,256}\\.[a-zA-Z0-9()]{1,6}\\b([-a-zA-Z0-9()@:%_\\+.~#?&//=]*)")
+  @NotBlank
+  @Size(max=255)
```
Comment (bkincart) [inline src/main/java/com/launchacademy/reviews/models/Venue.java:66]: Great validations!

**Review by bkincart — CHANGES_REQUESTED** [review]
Great work here! Some small updates requested :)

File: src/main/frontend/components/StateDropdown.js (hunk 6 above)
Comment (Jessica-Mach) [reply src/main/frontend/components/StateDropdown.js:65]: We added the potential refactor to a list on Trello and fixed the spacing.

**Pull request merged**